### PR TITLE
[#17435] migrate status-im.notifications

### LINKED ISSUE #17435: migrate status-im.notifications

### PR DESCRIPTION
**Pull request by flexsurfer**
migrate status-im.notifications

fixes: #17435

### CODE REVIEW

**Review by ibrkhalil — APPROVED** [review]

**Comment by pavloburykh** [comment]
@flexsurfer hi! Please take a look at the issue which causes all e2e to fail


### ISSUE 1 App crashes on sending/receiving CR (Android)

Reproduced on Android 12, Samsung Galaxy A52

**Steps:** 

1. Send CR from Device 1 to Device 2 (Android) or vice verse
2. Observe the result

**Actual result:** app crashes. Probably the problem is not only in CR but in generating notifs overall. Could not check other notifs because app keeps crashing after accepting CR, cannot even open 1-1 chat. 

[Status-debug-logs (69).zip](https://github.com/status-im/status-mobile/files/12858091/Status-debug-logs.69.zip)

photo_2023-10-10 17 05 40


https://github.com/status-im/status-mobile/assets/97245802/470ddf79-b476-4ce9-9879-4935e25dfe04

**Expected result:** no crash

**Comment by flexsurfer** [comment]
thanks @pavloburykh should be fixed now

**Review by Parveshdhull — APPROVED** [review]

File: src/status_im/ui/screens/notifications_settings/views.cljs [hunk 1]
```diff
@@ -166,6 +166,6 @@
         :after    :main-icon/next
         :disabled (empty? @server)
         :on-press #(do
-                     (re-frame/dispatch [::notifications/add-server @server])
+                     (re-frame/dispatch [:push-notifications/add-server @server])
```
Comment (siddarthkay) [inline src/status_im/ui/screens/notifications_settings/views.cljs]: nice

**Review by siddarthkay — APPROVED** [review]

**Comment by pavloburykh** [comment]
> thanks @pavloburykh should be fixed now

Hey @flexsurfer! 

Unfortunately crash is still reproducible

**Steps:** 

1. Device 1 (Android) sends CR to Device 2
2. Device 2 accepts CR
3. Device 1 (Android) enter new 1-1 chat with Device 2
4. Observe the result

**Actual result:** Device 1 (Android) crashes on opening 1-1 chat

**Comment by flexsurfer** [comment]
sorry @pavloburykh should be fixed now

**Comment by pavloburykh** [comment]
@flexsurfer thanx for the PR. Please take a look at the issues

### ISSUE 1 Notifications settings are disabled on IOS

Reproducing on IOS 16.1, iPhoneX

**Steps:** 

1. Enable notifications during onboarding
2. Go to Notifications settings page
3. See if notifications switcher is on. Try to interact with switcher
4. Observe the result

**Actual result:** notifications switcher is off despite notifications have been enabled. Cannot interact with switcher (it does not respond). At the same time PNs are in fact working. 


https://github.com/status-im/status-mobile/assets/97245802/96b80b25-564e-400a-a4fd-8e50008b4785


**Expected result:** user should be able to use old Notifications settings screen in order to manage notifications.

**Comment by pavloburykh** [comment]
### ISSUE 2 PNs generated on Desktop are not received by IOS device 

Reproducing on IOS 16.1, iPhoneX

**Preconditions:** User A (IOS) and User B (Desktop) share the same chat. User A (IOS) has PNs enabled. 

**Steps:** 

1. User B (Desktop) sends message to User A (IOS) in 1-1 chat (or mentions in any other chat)
2. See if User A (IOS) receives PN about new message from User B (Desktop) 

**Actual result:** no PN for User A. At the same time PNs from Desktop are received successfully on Android device. 

**Expected result:** User A receives PN. 

As Desktop version you can use Release 0.14.0 https://github.com/status-im/status-desktop/releases/tag/0.14.0

**Comment by flexsurfer** [comment]
thanks @pavloburykh 1 should be fixed, 2 for another issue

**Comment by pavloburykh** [comment]
@flexsurfer thanx for the fixes. PR is ready to be merged. 
BTW, [ISSUE 2](https://github.com/status-im/status-mobile/pull/17603#issuecomment-1759347395) is not reproducible any more too, I guess this is some random bug. Will log it in case if face it in future.

**Pull request merged**